frontend/ui/widget/virtualkeyboard.lua

```diff
@@ -32,14 +32,14 @@ local VirtualKey = InputContainer:new{
 }
 
 function VirtualKey:init()
-    if self.label == "Sym" or self.label == "ABC" then
-        self.callback = function () self.keyboard:setLayout(self.key or self.label) end
+    if self.label == "Sym" or self.label == "ABC" or self.label == "かな" or self.label == "記号" then
```

Comment:
Each keyboard layout could declare its own modifier keys, so the existing layouts would have 

- shiftmode_keys = {"Shift"}
- symbolmode_keys = {"Sym", "ABC"}
- utf8mode_keys = {"IM"}
- umlautmode_keys = {"Äéß"}

while the Japanese keyboard would have

- shiftmode_keys = {"Shift"}
- symbolmode_keys = {"記号", "かな"}
- utf8mode_keys = {"゛"}
- umlautmode_keys = {"゜"}

The callback setup in VirtualKey:init() would also have to switch on key instead of label, so that would have to be changed in the other layouts, too.